api/types/swarm/task.go

```diff
@@ -40,13 +40,14 @@ type Task struct {
 	Meta
 	Annotations
 
-	Spec                TaskSpec            `json:",omitempty"`
-	ServiceID           string              `json:",omitempty"`
-	Slot                int                 `json:",omitempty"`
-	NodeID              string              `json:",omitempty"`
-	Status              TaskStatus          `json:",omitempty"`
-	DesiredState        TaskState           `json:",omitempty"`
-	NetworksAttachments []NetworkAttachment `json:",omitempty"`
+	Spec                     TaskSpec            `json:",omitempty"`
+	ServiceID                string              `json:",omitempty"`
+	Slot                     int                 `json:",omitempty"`
+	NodeID                   string              `json:",omitempty"`
+	Status                   TaskStatus          `json:",omitempty"`
+	DesiredState             TaskState           `json:",omitempty"`
+	NetworksAttachments      []NetworkAttachment `json:",omitempty"`
+	AssignedGenericResources []*GenericResource  `json:",omitempty"`
```

Comment:
No you can't have unassigned resources that case do you think `GenericResources` is enough ?

It should not be a pointer type will change that